flink-runtime/src/test/java/org/apache/flink/runtime/state/OperatorStateBackendTest.java

```diff
@@ -418,6 +430,62 @@ void testSnapshotEmpty() throws Exception {
         assertThat(stateHandle).isNull();
     }
 
+    @Test
+    void testFileMergingSnapshotEmpty(@TempDir File tmpFolder) throws Exception {
```

Comment:
Could we also test the registery of the new handle ?
Or test that the subsumed checkpoint will discard correctly.

Reply:
I added `testSegmentStateHandleStateRegister` and `testFireMergingOperatorStateRegister` in `SharedStateRegistryTest`.